### PR TITLE
[DERCBOT-807] Add unit testing + coverage

### PR DESCRIPTION
**Pull request by GuirriecP**

### CODE REVIEW

File: llm/orchestrator-server/src/main/python/app/tests/test_exception_handlers.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+_request = httpx.Request('GET', 'https://www.dock.tock.ai')
+_response = httpx.Response(request=_request, status_code=400)
+
+def test_openai_exception_handler_api_connection_error():
+
+    @openai_exception_handler(provider='OpenAI or AzureOpenAIService')
+    def decorated_function(*args, **kwargs):
+        raise APIConnectionError(message='error', request=_request)
+    
+    try:
```
Comment (Benvii) [inline llm/orchestrator-server/src/main/python/app/tests/test_exception_handlers.py:46]: To test exceptions you should use pytest : 
```python
with pytest.raises(GenAIConnectionErrorException) as gen_exc:
    decorated_function()
assert gen_exc ......
```

See this documentation : https://docs.pytest.org/en/7.1.x/how-to/assert.html

File: llm/orchestrator-server/src/main/python/app/tests/test_langchain_factory.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+
+from unittest.mock import patch
+from llm_orchestrator.errors.exceptions.exceptions import GenAIUnknownProviderSettingException, VectorStoreUnknownException
+
+from llm_orchestrator.services.langchain.factories.langchain_factory import get_em_factory, get_llm_factory, get_vector_store_factory
+
+
+def test_get_llm_factory():
+    try:
```
Comment (Benvii) [inline llm/orchestrator-server/src/main/python/app/tests/test_langchain_factory.py:9]: Use pytest.raises

**Review by Benvii — CHANGES_REQUESTED** [review]
Main issue is to move files into different folders `integration`, `unit` tests at least.

File: llm/orchestrator-server/src/main/python/app/README.md [hunk 3]
```diff
@@ -112,6 +112,13 @@ pre-commit install
 uvicorn llm_orchestrator.main:app --host 0.0.0.0 --port 8000 --reload --log-level trace
 ```
 
+### Unit tests
+When dev dependencies are installed, run unit tests for the orchestrator using Tox:
+```
+tox run
+```
+This will also produce a Coverage.py code coverage report in coverage.xml.
```
[the remark this answers is not in the document]
Comment (GuirriecP) [reply llm/orchestrator-server/src/main/python/app/README.md]: Ajouté dans US à créer.

File: llm/orchestrator-server/src/main/python/app/README.md [hunk 4]
```diff
@@ -112,6 +112,13 @@ pre-commit install
 uvicorn llm_orchestrator.main:app --host 0.0.0.0 --port 8000 --reload --log-level trace
 ```
 
+### Unit tests
```
[the remark this answers is not in the document]
Comment (GuirriecP) [reply llm/orchestrator-server/src/main/python/app/README.md]: All dev dependencies are listed in poetry.lock, so 'poetry install' will install them along with the app.

File: llm/orchestrator-server/src/main/python/app/pyproject.toml [hunk 5]
```diff
@@ -21,6 +21,9 @@ colorlog = "^6.8.0"
 
 [tool.poetry.group.dev.dependencies]
```
[the remark this answers is not in the document]
Comment (GuirriecP) [reply llm/orchestrator-server/src/main/python/app/pyproject.toml]: Done

File: llm/orchestrator-server/src/main/python/app/tests/test_configurations.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+from llm_orchestrator.configurations.environement.settings import _Settings
+from llm_orchestrator.configurations.logging.logger import _setup_logging
+
+
+def test_environment():
```
[the remark this answers is not in the document]
Comment (GuirriecP) [reply llm/orchestrator-server/src/main/python/app/tests/test_configurations.py]: I don't understant the comment ;)

File: llm/orchestrator-server/src/main/python/app/tests/test_em_providers_router.py
[the remark this answers is not in the document]
Comment (GuirriecP) [reply llm/orchestrator-server/src/main/python/app/tests/test_em_providers_router.py:1]: The goal of the JIRA there was only to create unit tests, and tests using FastAPI TestClient are meant to be unit tests.
I understand these are not strictly speaking unit tests, but as we don't have functional testing at all, I think it is preferrable to keep them in place to have low-level functional coverage.

File: llm/orchestrator-server/src/main/python/app/tox.ini [hunk 7]
```diff
@@ -0,0 +1,16 @@
+[tox]
+envlist = py39
+skipsdist = True
+
+[testenv]
+allowlist_externals = poetry
+commands =
+    poetry install -v
```
[the remark this answers is not in the document]
Comment (GuirriecP) [reply llm/orchestrator-server/src/main/python/app/tox.ini]: Yes

File: llm/orchestrator-server/src/main/python/app/tests/test_exception_handlers.py (hunk 1 above)
Comment (GuirriecP) [reply llm/orchestrator-server/src/main/python/app/tests/test_exception_handlers.py]: Done.

File: llm/orchestrator-server/src/main/python/app/tests/test_langchain_factory.py (hunk 2 above)
Comment (GuirriecP) [reply llm/orchestrator-server/src/main/python/app/tests/test_langchain_factory.py]: Done

File: llm/orchestrator-server/src/main/python/app/tests/test_em_providers_router.py
[the remark this answers is not in the document]
Comment (Benvii) [reply llm/orchestrator-server/src/main/python/app/tests/test_em_providers_router.py:1]: DERCBOT-881

**Review by Benvii — APPROVED** [review]
LGTM

**Pull request merged**